Combined DKP 2.2 release notes draft 2 (#4206)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.2/release-notes/2.2.0/index.md

enterprise: false
beta: false
---

**D2iQ&reg; Kommander&reg; (DKP&reg;) version 2.2 was released on April 6, 2022.**

[button color="purple" href="https://support.d2iq.com/hc/en-us/articles/4409215222932-Product-Downloads"]Download Kommander[/button]

[Download](../download/) and [install](../install/) the latest version to get started.

<p class="message--note"><strong>NOTE: </strong>You must be a registered user and logged on to the support portal to download Kommander. New customers must contact their sales representative or <a href="mailto:sales@d2iq.com">sales@d2iq.com</a> before attempting to download or install this product.</p>

## Release summary

Welcome to D2iQ Kubernetes Platform (DKP) 2.2! This release continues our tradition of customer-led development making DKP even easier to deploy, supporting more platforms and applications, and includes our AI/ML solution, Kaptain, added as a catalog application. In this release, we are beginning the process of combining our two flagship products, Konvoy and Kommander, into a single DKP product with two service level options: DKP Enterprise for multi-cluster environments, and DKP Essentials for single-cluster environments.

> For this release, we are maintaining the legacy Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

Reviewer: - For this release, we are maintaining the legacy Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

We need to put this across in a better way. I don't like using legacy Konvoy and Kommander. Working with @Colin1841

> For this release, we are maintaining the legacy Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

Author: @sonali-mishra-22 - sure, we can remove that. We should have some way to differentiate between old way and new way of organizing things, but we can start by just removing "legacy" from here.
In general, do we want to think about how we refer to pre 2.2 products?

> For this release, we are maintaining the legacy Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

Author: ```suggestion
For this release, we maintain the Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.
```

> For this release, we are maintaining the legacy Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

Reviewer: We can say, for example:
`For this release, we are maintaining the documentation sets for platform components Konvoy and Kommander, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.
`
It's a lengthy process to remove all mentions of Konvoy and Kommander from docs, so let's not call them legacy. We can still talk about them as components.

> For this release, we are maintaining the legacy Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

Reviewer 2: I like that

This version supports Kubernetes versions between 1.21.0 and 1.22.x. Any cluster you want to attach using Kommander 2.2 must be running a Kubernetes version in this range.

## New features and capabilities

These features and capabilities are new for Version 2.2

> ### Upgrade catalog application via CLI and UI

Reviewer 2: I think we should list the upgrades overall as a feature and link to the upgrade docs.

Reviewer 2: @Colin1841 can we turn this catalog apps section into an upgrade section?

You can use either the CLI or the UI to [upgrade your catalog applications](../projects/applications/catalog-applications/#upgrade-catalog-applications).

<p class="message--note"><strong>NOTE:</strong> Catalog applications must be upgraded to the latest version BEFORE upgrading the Kubernetes version (or Konvoy version for managed Konvoy clusters) on attached clusters, due to the previous versions' incompatibility with Kubernetes 1.22.</p>

### Better integration with VMware vSphere

> You can use CAPI vSphere Provider while provisioning a DKP cluster on vSphere, which allows you to manage bootstrapping of VMs on a DKP cluster. This gives you improved productivity and speed of deploying VMs on DKP in a VMWare environment, including FIPS builds and air-gapped deployments.

Reviewer 2: Can we add a link to the vsphere docs?

### Zero downtime upgrades for air-gapped deployments

Kubernetes Operators and other IT operations team members can use your laptop or USB drive to transfer pre-created bundles, including OS dependencies and DKP binaries into an air-gapped environment with no external connectivity. This improves the availability of the DKP air-gapped deployment and productivity of your IT operations team.

### Unified user interfaces

This provides a smooth experience independent of where you start your journey. A DKP Essential customer can simply update your license to access DKP Enterprise features without having to learn another interface.

### Kaptain AI/ML, D2iQ’s AI/ML offering

> For better integrated with DKP 2.2 you can easily launch this feature as a catalog application, as well as support other platforms such as Amazon AWS, AKS, and Microsoft Azure EKS. This extends D2iQ’s openness to support Kubernetes platforms beyond DKP with Kaptain. Kaptain enables an organization to develop, deploy, and run entire ML workloads in production, at scale, with consistency and reliability.

Reviewer 2: ```suggestion
For better integration with DKP 2.2 you can launch Kaptain as a catalog application, as well as support other platforms such as Amazon AWS, AKS, and Microsoft Azure EKS. This extends D2iQ’s openness to support Kubernetes platforms beyond DKP with Kaptain. Kaptain enables an organization to develop, deploy, and run entire ML workloads in production, at scale, with consistency and reliability.
```

### DKP Insights

This new predictive analytics tool provides greater support productivity, speed, and reduced costs. An Insights module integrated into DKP collects configuration, logs, events, and metrics from DKP deployments and generates predictive insights on potential issues of varying critical levels. This enables you to quickly identify and resolve issues that would save a lot of time and money by not escalating to D2IQ's support. You can filter the insight summary cards by one or many insight categories for a selected cluster and project.

## Component updates

The following services and service components have been upgraded to the listed version:

> 

Author: @gracedo - from your example, and I pulled the apps and versions down from the 2.1.1 release notes, so these do need verified/updated, please. I left your example at the top, but we can move it down to match the alphabetical order here.

Author: @gvasquezvargas - FYI, here's the new table.

Reviewer 3: I will take care of updating this table in another PR that will merge into this one. Will use this jira to track https://jira.d2iq.com/browse/D2IQ-86410

<This is from 2.1.1, so we need to review and update. Grace has a good idea for an more user-friendly table here, to account for all versions and which versions of DKP everything synchs with. Will be easier to maintain here in the RN than buried in the full doc sets.>

> - centralized-grafana: 18.1.1

Reviewer 3: @gvasquezvargas we should replace this section with the table for app versions

---
layout: layout.pug

> navigationTitle: DKP 2.2 Release Notes

Reviewer 4: ```suggestion
navigationTitle: DKP 2.2.0 Release Notes
```

title: DKP 2.2 Release Notes
menuWeight: 10
excerpt: View release-specific information for DKP 2.2
enterprise: false
beta: false
---

**D2iQ&reg; Kommander&reg; (DKP&reg;) version 2.2 was released on April 6, 2022.**

[button color="purple" href="https://support.d2iq.com/hc/en-us/articles/4409215222932-Product-Downloads"]Download Kommander[/button]

[Download](../download/) and [install](../install/) the latest version to get started.

> <p class="message--note"><strong>NOTE: </strong>You must be a registered user and logged on to the support portal to download Kommander. New customers must contact their sales representative or <a href="mailto:sales@d2iq.com">sales@d2iq.com</a> before attempting to download or install this product.</p>

Reviewer 2: Ok yeah I think you are right. I changed the name of the button and put the sentence in there.

> <p class="message--note"><strong>NOTE: </strong>You must be a registered user and logged on to the support portal to download Kommander. New customers must contact their sales representative or <a href="mailto:sales@d2iq.com">sales@d2iq.com</a> before attempting to download or install this product.</p>

Reviewer 2: This is fixed.

For this release, we maintain the Konvoy and Kommander documentation sets, while publishing some combined DKP documentation for processes, such as Upgrading DKP version.

This version supports Kubernetes versions between 1.21.0 and 1.22.x. Any cluster you want to attach using Kommander 2.2 must be running a Kubernetes version in this range.

## New features and capabilities

These features and capabilities are new for Version 2.2

### Upgrade catalog application via CLI and UI

You can use either the CLI or the UI to [upgrade your catalog applications](../projects/applications/catalog-applications/#upgrade-catalog-applications).

<p class="message--note"><strong>NOTE:</strong> Catalog applications must be upgraded to the latest version BEFORE upgrading the Kubernetes version (or Konvoy version for managed Konvoy clusters) on attached clusters, due to the previous versions' incompatibility with Kubernetes 1.22.</p>

### Better integration with VMware vSphere

> You can use CAPI vSphere Provider while provisioning a DKP cluster on vSphere, which allows you to manage bootstrapping of VMs on a DKP cluster. This gives you improved productivity and speed of deploying VMs on DKP in a VMWare environment, including FIPS builds and air-gapped deployments.

Author: ```suggestion
You can use CAPI vSphere Provider while provisioning a [DKP cluster on vSphere](../../../choose-infrastructure/vsphere/index.md), which allows you to manage bootstrapping of VMs on a DKP cluster. This gives you improved productivity and speed of deploying VMs on DKP in a VMWare environment, including FIPS builds and air-gapped deployments.
```

Reviewer: @Colin1841 We should make it:
which allows you to manage bootstrapping of VMs for/in a DKP cluster.
DKP cluster is on a VM, not the other way round.

Reviewer 5: Wouldn't the link for Kommander need to cross over to the Konvoy docs?

Author: Doh, yes - need to link to Kon docs. Do we need an actual link to that, vs relative link?

> You can use CAPI vSphere Provider while provisioning a DKP cluster on vSphere, which allows you to manage bootstrapping of VMs on a DKP cluster. This gives you improved productivity and speed of deploying VMs on DKP in a VMWare environment, including FIPS builds and air-gapped deployments.

Reviewer 5: ```suggestion
You can use CAPI vSphere Provider while [provisioning a DKP cluster on vSphere][vsphere], which allows you to manage bootstrapping of VMs on a DKP cluster. This gives you improved productivity and speed of deploying VMs on DKP in a VMWare environment, including FIPS builds and air-gapped deployments.
```

### Zero downtime upgrades for air-gapped deployments

Kubernetes Operators and other IT operations team members can use your laptop or USB drive to transfer pre-created bundles, including OS dependencies and DKP binaries into an air-gapped environment with no external connectivity. This improves the availability of the DKP air-gapped deployment and productivity of your IT operations team.

### Unified user interfaces

> This provides a smooth experience independent of where you start your journey. A DKP Essential customer can simply update your license to access DKP Enterprise features without having to learn another interface.

Author: @cneth I pulled this verbiage from the Marketing draft, that Mkt and Prod teams jointly authored, so I'm not sure either!

> This provides a smooth experience independent of where you start your journey. A DKP Essential customer can simply update your license to access DKP Enterprise features without having to learn another interface.

Reviewer 2: let's just remove some of this then

> This provides a smooth experience independent of where you start your journey. A DKP Essential customer can simply update your license to access DKP Enterprise features without having to learn another interface.

Reviewer 2: fixed

### Kaptain AI/ML, D2iQ’s AI/ML offering

> For better integration with DKP 2.2 you can launch Kaptain as a catalog application, as well as support other platforms such as Amazon AWS, AKS, and Microsoft Azure EKS. This extends D2iQ’s openness to support Kubernetes platforms beyond DKP with Kaptain. Kaptain enables an organization to develop, deploy, and run entire ML workloads in production, at scale, with consistency and reliability.

Reviewer 2: fixed

### DKP Insights

> This new predictive analytics tool provides greater support productivity, speed, and reduced costs. An Insights module integrated into DKP collects configuration, logs, events, and metrics from DKP deployments and generates predictive insights on potential issues of varying critical levels. This enables you to quickly identify and resolve issues that would save a lot of time and money by not escalating to D2IQ's support. You can filter the insight summary cards by one or many insight categories for a selected cluster and project.

Reviewer 6: This needs to be reworked, we should borrow from the insights PR here

Reviewer 2: I modified this some, but maybe needs more?

### Zero downtime upgrades for air-gapped deployments

You can now use your laptop or USB drive to transfer pre-created air-gapped bundles, including OS dependencies and DKP binaries into your air-gapped environment with no external connectivity. This improves the availability of the DKP air-gapped deployment and productivity of your IT operations team.

For more information, see the [air-gapped bundle](????????) documentation. 

### Unified DKP user interfaces

The unified DKP user interface provides a smooth experience independent of where you start your journey. A DKP Essential customer can simply update your license to access DKP Enterprise features without having to learn another interface.

### Kaptain AI/ML, D2iQ’s AI/ML offering

For better integration with DKP 2.2 you can launch Kaptain as a catalog application, and also support other platforms such as Amazon AWS, AKS, and Microsoft Azure EKS. This extends D2iQ’s openness to support Kubernetes platforms beyond DKP with Kaptain. Kaptain enables an organization to develop, deploy, and run entire ML workloads in production, at scale, with consistency and reliability.

### DKP Insights

> This new predictive analytics tool provides greater support productivity, speed, and reduced costs. An [Insights](../../insights/) module integrated into DKP collects configuration, logs, events, and metrics from DKP deployments and generates predictive insights on potential issues of varying levels. These insights enable you to proactively identify and resolve issues that can save you a lot of time and money by not escalating to D2iQ's support. 

Author: ```suggestion
This new predictive analytics tool provides greater support productivity, speed, and reduced costs. The [DKP Insights](../../insights/) Engine collects events and metrics on the Attached cluster, and uses rule-based heuristics on potential problems of varying criticality, so they can be quickly identified and resolved. These Insights are then forwarded and displayed in the DKP Insights Dashboard. It assists you with routine tasks such as :

- Resolving common issues
- Monitoring resource usage
- Checking security issues
- Verifying workloads and clusters follow best practices

```

Author: @cneth - how's this for aligning a little better with the Insights doc pages?

---
layout: layout.pug
navigationTitle: DKP 2.2.0 Release Notes
title: DKP 2.2 Release Notes
menuWeight: 10
excerpt: View release-specific information for DKP 2.2.0
enterprise: false
beta: false
---

**D2iQ&reg; Kommander&reg; (DKP&reg;) version 2.2 was released on April 6, 2022.**

[button color="purple" href="https://support.d2iq.com/hc/en-us/articles/4409215222932-Product-Downloads"]Download DKP[/button]

> **Note:** In DKP 2.2, the Konvoy and Kommander binaries have been merged into a single binary, which you can find by selecting the DKP button above. 

Author: ```suggestion
**Note:** In DKP 2.2 the Konvoy and Kommander binaries have been merged into a single binary, which you can find by selecting the DKP button above. 
```
